src/components/tables/AccessGrants/AccessLinks/Dialog/GenerateInvitation.tsx

```diff
@@ -91,82 +163,218 @@ function GenerateInvitation({
         >
-            <Grid
-                item
-                xs={12}
-                sx={showNestingWarning ? { mb: 1 } : { display: 'none' }}
-            >
-                <NestingWarning
-                    objectRole={objectRole}
-                    show={showNestingWarning}
-                />
-            </Grid>
-
-            <Grid item xs={12} md={5} sx={{ display: 'flex' }}>
-                <PrefixedName
-                    allowBlankName
-                    allowEndSlash
-                    defaultPrefix
-                    label={intl.formatMessage({
-                        id: 'common.tenant',
-                    })}
-                    onChange={onChange}
-                    onNameChange={(prefixedName) => {
-                        setShowNestingWarning(
-                            Boolean(prefixedName && prefixedName.length > 0)
-                        );
+            <Box sx={{ display: 'flex', gap: 2 }}>
+                <Box
+                    sx={{
+                        display: 'flex',
+                        flexDirection: 'column',
+                        gap: 2,
+                        minWidth: 180,
                     }}
-                    required
-                    validateOnLoad
-                />
-            </Grid>
+                >
+                    <Box sx={{ pt: 1 }}>
+                        <PrefixedName
+                            prefixOnly
+                            defaultPrefix
+                            disabled={objectRoles?.length === 1}
+                            label={intl.formatMessage({
+                                id: 'admin.users.prefixInvitation.label.tenant',
+                            })}
+                            onChange={onChange}
+                        />
+                    </Box>
 
-            <Grid item xs={4} md={2}>
-                <AutocompletedField
-                    label={intl.formatMessage({
-                        id: 'admin.users.prefixInvitation.label.capability',
-                    })}
-                    required
-                    options={capabilityOptions}
-                    defaultValue={capabilityOptions[0]}
-                    changeHandler={handlers.setGrantCapability}
-                />
-            </Grid>
+                    <AutocompletedField
+                        label={intl.formatMessage({
+                            id: 'admin.users.prefixInvitation.label.capability',
+                        })}
+                        required
+                        options={capabilityOptions}
+                        defaultValue={capabilityOptions[0]}
+                        changeHandler={handlers.setGrantCapability}
+                    />
+                </Box>
+
+                <Divider flexItem orientation="vertical" />
 
-            <Grid item xs={4} md={2}>
-                <AutocompletedField
+                <Box sx={{ flex: 1 }}>
+                    <RadioGroup
```

Comment:
This is an interesting thing we'll wanna chat about in dailies on Monday.

Up until now normally selecting from a group of options like this go into a raio drop down. We could use some of these styles and approaches... however an actual menu here would be a big step back I think.
<img width="438" height="461" alt="Image" src="https://github.com/user-attachments/assets/b13afa6e-cce3-4efe-b088-25fa70fb87ea" />